File: src/controller/dto/incoming/newChore.dto.ts
import { Type } from 'class-transformer';
import {
  IsDate,
  IsInt,
  IsOptional,
  IsPositive,
  IsString,
  Length,
  MaxLength,
  MinDate,
} from 'class-validator';

export class NewChoreDTO {
  @IsString()
  @Length(1, 50)
  title!: string;

  @IsOptional()
  @IsString()
  @MaxLength(255)
  description!: string;

  @IsDate()
  @MinDate(new Date())
  @Type(() => Date)
  deadline!: Date;

  @IsInt()
  @IsPositive()
  priorityLevel!: number;
}

File: bnim2219-spring/src/main/java/edu/bbte/idde/bnim2219/spring/controller/dto/incoming/NewChoreDTO.java
package edu.bbte.idde.bnim2219.spring.controller.dto.incoming;

import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;

import java.sql.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class NewChoreDTO {
    @NotNull
    @NotEmpty
    @Length(max = 50)
    private String title;
    @Length(max = 255)
    private String description;
    @NotNull
    @FutureOrPresent
    private Date deadline;
    @NotNull
    @Positive
    private Integer priorityLevel;
}
